test/functional/api/v2/hosts_controller_test.rb

```diff
@@ -605,4 +605,20 @@ def initialize_proxy_ops
     get :template, { :id => managed_host.to_param, :kind => 'provitamin' }
     assert_response :not_found
   end
+
+  test "should search host by hostgroup name" do
+    hostgroup = FactoryGirl.create(:hostgroup, :with_parent)
+    managed_host = FactoryGirl.create(:host, :managed, :hostgroup => hostgroup)
+    get :index, {:search => "hostgroup_name = #{hostgroup.name}"}
+    hosts = ActiveSupport::JSON.decode(@response.body)['results']
```

Comment:
You can `assert_equal [managed_host], assigns(:hosts)` instead of parsing the response, which is a more expensive operation